src/batchproof_container.cpp

```diff
@@ -159,39 +160,70 @@ void BatchProofContainer::erase(std::vector<LelantusSigmaProofData>* vProofs, co
 void BatchProofContainer::batch_sigma() {
     if (!sigmaProofs.empty())
         LogPrintf("Sigma batch verification started.\n");
-    for (const auto& itr : sigmaProofs) {
-        std::vector<GroupElement> anonymity_set;
-        sigma::CSigmaState* sigmaState = sigma::CSigmaState::GetState();
-        sigmaState->GetAnonymitySet(
-                itr.first.first,
-                itr.first.second.first,
-                itr.first.second.second,
-                anonymity_set);
-
-        size_t m = itr.second.size();
-        std::vector<Scalar> serials;
-        serials.reserve(m);
-        std::vector<bool> fPadding;
-        fPadding.reserve(m);
-        std::vector<size_t> setSizes;
-        setSizes.reserve(m);
-        std::vector<sigma::SigmaPlusProof<Scalar, GroupElement>> proofs;
-        proofs.reserve(m);
-
-        for (auto& proofData : itr.second) {
-            serials.emplace_back(proofData.coinSerialNumber);
-            fPadding.emplace_back(proofData.fPadding);
-            setSizes.emplace_back(proofData.anonymitySetSize);
-            proofs.emplace_back(proofData.sigmaProof);
-        }
+    else
+        return;
+    std::size_t threadsMaxCount = std::min((unsigned int)sigmaProofs.size(), boost::thread::hardware_concurrency());
+    std::vector<boost::future<bool>> parallelTasks;
+    parallelTasks.reserve(threadsMaxCount);
+    ParallelOpThreadPool<bool> threadPool(threadsMaxCount);
```

Comment:
DoNotDisturb here? and everywhere where ParallelOpThreadPool is used

Reply:
We added multithreading in Lelantus-sigma roof creation code and batch proving code.
1 Added DoNotDisturb here for proof creation, I think we need it inside the function which is called in thread. https://github.com/firoorg/firo/pull/1058/files#diff-35b15f4d3b021c6dc41983092fa0d3666801234fe9f9ce790b41c65297a68483R32
2 Fir batching I don't think we need don't disturb as shutdown waits until batch verification finishes, [ ](https://github.com/firoorg/firo/blob/78e29d68b7354be702965fdd4f033709750776e5/src/init.cpp#L258)